pipe-cli/pipe.py

```diff
@@ -1226,42 +1227,82 @@ def ssh(ctx, run_id, retries):
               help='Establishes tunnel in foreground mode')
 @click.option('-u', '--user', required=False, callback=set_user_token, expose_value=False, help=USER_OPTION_DESCRIPTION)
+@click.option('-r', '--retries', required=False, type=int, default=10, help=RETRIES_OPTION_DESCRIPTION)
+@click.option('--trace', required=False, is_flag=True, default=False, help=TRACE_OPTION_DESCRIPTION)
 @Config.validate_access_token
-def tunnel(run_id, local_port, remote_port, log_file, log_level, timeout, foreground):
+def tunnel(run_id, local_port, remote_port, connection_timeout,
+           ssh, ssh_path, ssh_host, ssh_keep, log_file, log_level,
+           timeout, foreground, retries, trace):
     """
     Establishes tunnel connection to specified run instance port and serves it as a local port.
 
     It allows to transfer any tcp traffic from local machine to run instance and works both on Linux and Windows.
 
-    For example it can be used to allow ssh connections to run instance.
+    Additionally it enables passwordless ssh connections if the corresponding option is specified.
+    Once specified ssh is configured both locally and remotely to support passwordless connections.
+    Passwordless ssh configuration is supported only for openssh client on Linux.
+
+    Examples:
+
+    I. Example of simple tcp port tunnel connection establishing.
+
+    Establish tunnel connection from run (12345) instance port (4567) to the same local port.
+
+        pipe tunnel -lp 4567 -rp 4567 12345
+
+    II. Example of ssh port tunnel connection establishing with enabled passwordless ssh configuration.
 
-    \b
     First of all establish tunnel connection from run (12345) instance ssh port (22) to some local port (4567).
-        pipe tunnel -lp 4567 -rp 22 12345
 
-    \b
-    Then connect to run instance using regular ssh client on the configured local port (4567).
-        ssh -p 4567 root@localhost
+        pipe tunnel -lp 4567 -rp 22 --ssh 12345
+
+    Then connect to run instance using regular ssh client.
+
+        ssh root@pipeline-12345
+
+    Or transfer some files to and from run instance using regular scp client.
+
+        scp file.txt root@pipeline-12345:/common/workdir/file.txt
+
+    Advanced tunnel configuration environment variables:
 
     \b
-    Additionally the following environment variables can be used to specify the exact tunnel properties.
-        CP_CLI_TUNNEL_PROXY_HOST
-        CP_CLI_TUNNEL_PROXY_PORT
-        CP_CLI_TUNNEL_TARGET_HOST
+        CP_CLI_TUNNEL_PROXY_HOST - tunnel proxy host
+        CP_CLI_TUNNEL_PROXY_PORT - tunnel proxy port
+        CP_CLI_TUNNEL_TARGET_HOST - tunnel target host
+        CP_CLI_TUNNEL_SERVER_ADDRESS - tunnel server address
     """
-    try:
-        create_tunnel(run_id, local_port, remote_port, log_file, log_level, timeout, foreground)
-    except Exception as runtime_error:
-        click.echo('Error: {}'.format(str(runtime_error)), err=True)
-        sys.exit(1)
+    if trace:
```

Comment:
If the goal is to print stack trace we can use something like `traceback.print_exc()` in `except` section